users/vieting/experiments/switchboard/hybrid/feat/specaug_sorted.py

```diff
@@ -0,0 +1,153 @@
+
+def sort_filters_by_center_freq(x, return_sorted_filters=True):
+    """
+    This function either sorts the filters by their center frequency and returns them,
+    or it returns the indices that would sort the filters.
+    :param tf.Tensor x: The filter layer to sort.
+    :param bool return_sorted_filters: Whether to return the sorted filters or the sorted indices.
+    :return: Sorted filters or sorted indices.
+    """
+    x = tf.convert_to_tensor(x)  # (256, 1, 150) = (N, 1, C)
+  
+    # implementation similar to scipy.signal.freqz, which uses numpy.polynomial.polynomial.polyval
+    filters = tf.transpose(tf.squeeze(x))  # (C, N)
+    num_freqs = 512  # F
+    w = tf.linspace(0.0, np.pi - np.pi / num_freqs, num_freqs)  # (F,)
+    zm1 = tf.expand_dims(tf.exp(-1j * tf.cast(w, "complex64")), 1)  # (F, 1)
+    exponents = tf.expand_dims(tf.range(tf.shape(filters)[1]), 0)  # (1, N)
+    zm1_pow = tf.pow(zm1, tf.cast(exponents, dtype="complex64"))  # (F, N)
+    f_resp = tf.tensordot(zm1_pow, tf.cast(tf.transpose(filters), dtype="complex64"), axes=1)  # (F, C)
+    f_resp = tf.abs(f_resp)
+    # move to log domain, not needed for center frequencies
+    # f_resp = 20 * tf.math.log(f_resp) / tf.math.log(tf.constant(10.0))
+
+    # sorted by increasing center frequency
+    center_freqs = tf.argmax(f_resp, axis=0)
+    sorted_idcs = tf.argsort(center_freqs)
+
+    if return_sorted_filters:
+        x = tf.gather(x, sorted_idcs, axis=2)
+        return x
+    else:
+        return sorted_idcs
+
+
+
+def _mask(x, batch_axis, axis, pos, max_amount, sorted_indices=None):
+        """
+        :param tf.Tensor x: (batch,time,feature)
+        :param int batch_axis:
+        :param int axis:
+        :param tf.Tensor pos: (batch,)
+        :param int|tf.Tensor max_amount: inclusive
+        :param tf.Tensor|None sorted_indices: sorted indices (optional)
+        """
+        ndim = x.get_shape().ndims
+        n_batch = tf.shape(x)[batch_axis]
+        dim = tf.shape(x)[axis]
+        amount = tf.random.uniform(shape=(n_batch,), minval=max_amount, maxval=max_amount + 1, dtype=tf.int32)
```

Comment:
```suggestion
        amount = tf.random_uniform(shape=(n_batch,), minval=1, maxval=max_amount + 1, dtype=tf.int32)
```
This deviates from the original specaugment implementation. In particular, the different `minval` can only be used for debugging and would be critical here. Please do a vimdiff with the original file to fix differences, that are not intended and needed for the sorting. I'll review the rest of the file after that.